### ISSUE TITLE
Active Training samples both Node.js and C# seem to have logic errors

### ISSUE THREAD

**Issue by petmat** [Microsoft/BotBuilder-Samples#1204]
## Sample information
1. Sample type: \samples\experimental\
2. Sample language: nodejs/c#
3. Sample name: qnamaker_activelearning-bot

## Describe the bug
In the `getLowScoreVariation function when the `isSingleAnswerSuggestion` function is called the `qnaSearchResults[0].score` value is used without multiplying it with 100 like elsewhere in the code. This means that it would only return false if the score was more than 5000 %. 

In the Node.js version the line after calling `isSingleAnswerSuggestion` returns an object instead of an array which is inconsistent with other return values of the function.

It i really hard to make out what the function is supposed to do. It could use some more comments. Could you clarify why for example if there is a single QnA search result with a score higher than 95% an empty array is returned but with a single QnA search result score of 40% an array with one item is returned? This makes no difference in the call site of the `getLowScoreVariation` function since there only `filteredResponses.length > 1` is compared but it is hard to follow the logic here.

**Comment by rokulka** [comment]
Thanks for reaching out @petmat 

The isSingleAnswerSuggestion was actually not used, if you look at FilterLowVariationScoreList in DialogHelper. We check  if (filteredResponse.Count > 1).  I've cleaned up that part of the code in the above PR. Sorry for the confusion. 

Regarding the low score variation logic - For each step in traversing the qna list, the window width is currently a function of square root of the current qna score tuned with a multiplier. The parameters are fixed based on what suits best with the scores returned by our ranker.
